### PR TITLE
[SAIC-280] Grouping may produce invalid results if tenant has multiple networks with the same name

### PR DESCRIPTION
**Pull request by Settis**
Now grouping based on network id.

### CODE REVIEW

**Comment by Wasapl** [comment]
Build finished.
Unittest failed

**Comment by Wasapl** [comment]
Unittest failed

**Comment by Wasapl** [comment]
Build finished.
Functional test failure

**Comment by Wasapl** [comment]
Functional test failure

**Comment by miarmak** [comment]
1 small note: Please, make commit message shorter (<=50) and more valuable to have commit history readable. You can put more text if you need to the commit description below (<=72).

File: cloud/grouping.py [hunk 1]
```diff
@@ -90,14 +90,20 @@ def _group_nested_tenant(self, instances_list=None):
...
                        self.compute.get_instances_list(
                            search_opts={"all_tenants": True}))
-        for network in network_list:
-            LOG.info('Processing network %s', network['name'])
-            groups[str(network['name'])] = [vm for vm in search_list
-                                            if network['name'] in vm.networks]
+        for instance in search_list:
+            LOG.info('Processing instance %s', instance.name)
+            for (network_name, network_ips) in instance.networks.items():
+                network = self.network.get_network({'ip': network_ips[0]},
```
Comment (roman-verchikov) [inline cloud/grouping.py]: This doesn't feel right, previous version added VM to all groups of VMs based on network name, this pattern should be followed here too.

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

**Comment by Wasapl** [comment]
Build finished.
Functional test success

**Comment by Wasapl** [comment]
Functional test success

File: cloud/grouping.py (hunk 1 above)
Comment (Settis) [inline cloud/grouping.py]: Yes. I follow this rule. The result map is 'network_id' => 'instance'.
But I iterates by instance. I think it's doesn't matter.

**Comment by roman-verchikov** [comment]
+1

**Comment by roman-verchikov** [comment]
Functional tests succeed, can be merged

**Pull request merged**